### PR TITLE
Add check for `target_sizes is None` in `post_process_image_guided_detection` for owlv2

### LINKED ISSUE #31915: `target_sizes` in Owlv2 `post_process_image_guided_detection`
### System Info

N/A

### Who can help?

_No response_

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

In `post_process_image_guided_detection` ([link](https://github.com/huggingface/transformers/blob/main/src/transformers/models/owlv2/image_processing_owlv2.py#L544C9-L544C44)), there's no check for when `target_sizes = None`, so calling it without providing `target_sizes` raises an error.

### Expected behavior

The case where `target_sizes` is `None` should be handled, as done in `post_process_detection` ([link](https://github.com/huggingface/transformers/blob/main/src/transformers/models/owlv2/image_processing_owlv2.py#L484))

### PR DESCRIPTION
**Pull request by catalys1**
Updates the `Owlv2ImageProcessor.post_process_image_guided_detection` by adding check for when `target_sizes is None`, as is done in the analogous function `post_process_detection`.

Fixes #31915 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by qubvel** [comment]
Hi @catalys1,
Thanks a lot for reporting the issue and submitting the fix! 🤗 

Can you, please, make a few more steps to finish this PR?

1) Method, modified in PR is a copy of the method from OwlViTImageProcessor, can you please make the same modifications to the original method and then run `make fix-copies` to make sure both methods are consistent?

On the top of the method you can see:
```python
# Copied from transformers.models.owlvit.image_processing_owlvit.OwlViTImageProcessor.post_process_image_guided_detection
```

2) run `make modified_only_fixup` - this will check  formatting and fix quality errors 

Thanks!

**Comment by catalys1** [comment]
Done

File: src/transformers/models/owlv2/image_processing_owlv2.py [hunk 1]
```diff
@@ -565,10 +565,11 @@ def post_process_image_guided_detection(self, outputs, threshold=0.0, nms_thresh
...
 
-        if len(logits) != len(target_sizes):
-            raise ValueError("Make sure that you pass in as many target sizes as the batch dimension of the logits")
-        if target_sizes.shape[1] != 2:
-            raise ValueError("Each element of target_sizes must contain the size (h, w) of each image of the batch")
+        if target_sizes is not None:
+            if len(logits) != len(target_sizes):
+                raise ValueError(
+                    "Make sure that you pass in as many target sizes as the batch dimension of the logits"
+                )
```
Comment (amyeroberts) [inline src/transformers/models/owlv2/image_processing_owlv2.py:572]: nit - if just this check, then we can combine the two if statements
```suggestion
        if target_sizes is not None and len(logits) != len(target_sizes):
            raise ValueError(
                "Make sure that you pass in as many target sizes as the batch dimension of the logits"
            )
```

Is there a reason for removing the second check? 


```suggestion
        if target_sizes is not None and len(logits) != len(target_sizes):
            raise ValueError("Make sure that you pass in as many target sizes as the batch dimension of the logits")
        if target_sizes is not None and target_sizes.shape[1] != 2:
            raise ValueError("Each element of target_sizes must contain the size (h, w) of each image of the batch")
```

File: src/transformers/models/owlvit/image_processing_owlvit.py [hunk 2]
```diff
@@ -556,10 +556,11 @@ def post_process_image_guided_detection(self, outputs, threshold=0.0, nms_thresh
         """
         logits, target_boxes = outputs.logits, outputs.target_pred_boxes
 
-        if len(logits) != len(target_sizes):
```
Comment (amyeroberts) [inline src/transformers/models/owlvit/image_processing_owlvit.py:559]: Same here

File: src/transformers/models/owlv2/image_processing_owlv2.py [hunk 3]
```diff
@@ -588,7 +589,12 @@ def post_process_image_guided_detection(self, outputs, threshold=0.0, nms_thresh
                     scores[idx][ious > nms_threshold] = 0.0
 
         # Convert from relative [0, 1] to absolute [0, height] coordinates
-        img_h, img_w = target_sizes.unbind(1)
+        if target_sizes is not None:
```
Comment (amyeroberts) [inline src/transformers/models/owlv2/image_processing_owlv2.py:592]: if `target_sizes` is `None` we still need to set img_h, img_w

**Review by amyeroberts — COMMENTED** [review]
Thanks for working on this!

File: src/transformers/models/owlv2/image_processing_owlv2.py (hunk 1 above)
Comment (catalys1) [reply src/transformers/models/owlv2/image_processing_owlv2.py:572]: I basically just changed it to match [`post_process_object_detection`](https://github.com/huggingface/transformers/blob/main/src/transformers/models/owlv2/image_processing_owlv2.py#L506). But the edits look fine to me.

File: src/transformers/models/owlv2/image_processing_owlv2.py (hunk 3 above)
Comment (catalys1) [reply src/transformers/models/owlv2/image_processing_owlv2.py:592]: I thought if `target_sizes` is `None`, then we don't scale the bounding boxes, we leave them in relative coordinates. In which case, we don't need to go anything related to scaling.

Comment (amyeroberts) [reply src/transformers/models/owlv2/image_processing_owlv2.py:592]: Right, but the line on L589 expects the img_w and img_h variables, which will now errro out as they're undefined if target_sizes is None. I suspect L598 and L599 should also be indented

File: src/transformers/models/owlvit/image_processing_owlvit.py [hunk 4]
```diff
@@ -579,7 +580,12 @@ def post_process_image_guided_detection(self, outputs, threshold=0.0, nms_thresh
...
 
         # Convert from relative [0, 1] to absolute [0, height] coordinates
-        img_h, img_w = target_sizes.unbind(1)
+        if target_sizes is not None:
+            if isinstance(target_sizes, List):
+                img_h = torch.tensor([i[0] for i in target_sizes])
+                img_w = torch.tensor([i[1] for i in target_sizes])
+            else:
+                img_h, img_w = target_sizes.unbind(1)
         scale_fct = torch.stack([img_w, img_h, img_w, img_h], dim=1).to(target_boxes.device)
```
Comment (amyeroberts) [inline src/transformers/models/owlvit/image_processing_owlvit.py:589]: There's the same issue here re img_w and img_h being defined

File: src/transformers/models/owlv2/image_processing_owlv2.py (hunk 3 above)
Comment (catalys1) [reply src/transformers/models/owlv2/image_processing_owlv2.py:592]: Yes, they should be inside the if block. That's my mistake.

**Review by amyeroberts — APPROVED** [review]
Thanks for updating and iterating on this @catalys1!

**Pull request merged**